File: CandidateApp.Domain/Models/Candidate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CandidateApp.Domain.Models
{
    public class Candidate
    {
        public int ID { get; set; }
        public string FirstName { get; set; }
        public string Surname { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string Address1 { get; set; }
        public string Town { get; set; }
        public string Country { get; set; }
        public string PostCode { get; set; }
        public string PhoneHome { get; set; }
        public string PhoneMobile { get; set; }
        public string PhoneWork { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
    }
}

File: CandidateApp.Client/src/app/models/candidate.ts
import { Skill } from "./skill";

export class Candidate {
  id: number;
  firstName: string;
  surname: string;
  dateOfBirth: Date;
  address1: string;
  town: string;
  country: string;
  postCode: string;
  phoneHome: string;
  phoneMobile: string;
  phoneWork: string;
  createdDate: Date;
  updatedDate: Date;
  skills: Skill[]

  constructor(
    id: number,
    firstName: string,
    surname: string,
    dateOfBirth: Date,
    address1: string,
    town: string,
    country: string,
    postCode: string,
    phoneHome: string,
    phoneMobile: string,
    phoneWork: string,
    createdDate: Date,
    updatedDate: Date,
    skills: Skill[]
  ) {
    this.id = id;
    this.firstName = firstName;
    this.surname = surname;
    this.dateOfBirth = dateOfBirth;
    this.address1 = address1;
    this.town = town;
    this.country = country;
    this.postCode = postCode;
    this.phoneHome = phoneHome;
    this.phoneMobile = phoneMobile;
    this.phoneWork = phoneWork;
    this.createdDate = createdDate;
    this.updatedDate = updatedDate;
    this.skills = skills;
  }
}
